profiling/src/internal/profile.rs

```diff
@@ -544,6 +544,188 @@ impl Profile {
 #[cfg(test)]
 mod api_tests {
     use super::*;
+    use bolero::TypeGenerator;
+    use itertools::Itertools;
+    use std::collections::HashSet;
+
+    /// Fuzzes adding a bunch of samples to the profile.
+    /// TODO: many of the samples will be of the wrong length, and will give a boring error right
+    /// away.  Improve the generator to test the case where samples are the right length to get
+    /// more coverage faster
+    /// TODO: Test the expected result of adding the samples
+    #[test]
+    fn fuzz_add_sample() {
+        bolero::check!()
+            .with_generator((
+                Vec::<owned_types::ValueType>::gen(),
+                Vec::<(Option<Timestamp>, owned_types::Sample)>::gen(),
+            ))
+            .for_each(|(val, samples)| {
+                let expected_sample_types: Vec<_> = val.iter().map(api::ValueType::from).collect();
+                let mut expected_profile =
+                    Profile::new(SystemTime::now(), &expected_sample_types, None);
+                let mut expected_samples: Vec<&owned_types::Sample> = Vec::new();
```

Comment:
This will have an issue with aggregated samples

Reply:
Yes, updated the code to keep track of timestamped and non-timestamped samples.